### DISCUSSION TITLE
osu!lazer crashes upon pressing "Save" in skin editor

### DISCUSSION THREAD

**Discussion by Morilli** [ppy/osu#14884] [Q&A] [answered]
I open a random map, pause, open skin editor, press save. I can see a database unique constraint error in the debugger and have no idea what it means. *Note that the game breaks upon pressing save once, but only crashes after pressing it again.*

osu!lazer version: `2021.927.0`

**Comment by frenzibyte** [comment]
Please attach game logs.

**Reply by Morilli** [reply]
[runtime.log](https://github.com/ppy/osu/files/7247764/runtime.log)
Seems to not happen when using the default osu!lazer or osu!stable skins, so here's the link for my skin: https://drive.google.com/file/d/1-MY-jJgOSvi8X1_oC_xDE4uq3V79MN1C/view?usp=sharing

**Reply by peppy** [reply]
If possible could you please shared your `client.db` file? It should not have any confidential information so is safe to post publicly.

**Reply by Morilli** [reply]
[client.db.zip](https://github.com/ppy/osu/files/7253898/client.db.zip)

**Reply by bdach** [reply]
@Morilli in your database you happen to have two similar skins: one named `Clean osu!+mania`, and one named `Clean osu!+mania [main skin]`. did you perhaps edit one of them previously, and then tried to edit the other and got the above crash?

**Reply by Morilli** [reply]
I have this skin twice, but once in a kind of "backup" version and a newer version (the `main skin` one). The skin.ini is the exact same, but the folder name is different (and contents of course).

**Reply by bdach** [reply]
the `skin.ini` files being identical is the root cause of the issue, as the hash of the skin is based on it:

https://github.com/ppy/osu/blob/8bbd8cd948c7c903e05c8e11cc5d54275926341c/osu.Game/Skinning/SkinManager.cs#L141-L150

having two skins with the same hash is tripping the unique constraint.

**Reply by peppy** [reply]
That was my thinking. Question is how the two skins were imported in the first place.

**Reply by GiantNuker** [reply]
> That was my thinking. Question is how the two skins were imported in the first place.

Importing from stable, I did that and had the same problem an hour ago

**Comment by Wieku** [comment]
Seems like a duplicate of #14850

**Comment by peppy** [comment, marked as answer]
Will be fixed by #15204.
